server/src/main/java/org/opensearch/ingest/ConfigurationUtils.java

```diff
@@ -510,9 +511,11 @@ public static Processor readProcessor(
         Map<String, Processor.Factory> processorFactories,
         ScriptService scriptService,
         String type,
-        Object config
+        @Nullable Object config
     ) throws Exception {
-        if (config instanceof Map) {
+        if (config == null) {
+            throw newConfigurationException(type, null, null, "processor [" + type + "] cannot be null");
```

Comment:
```suggestion
            throw newConfigurationException(type, null, null, "the config of processor [" + type + "] cannot be null");
```
(since we checking the config, not processor)